src/DropdownMenu/DropdownMenu.tsx

```diff
@@ -0,0 +1,158 @@
+   */
+  selectedItem?: ItemInput
+
+  /**
+   * A callback which receives the selected item or `undefined` when an item is activated in the menu.  If the activated item is the same as the current
+   * `selectedItem`, `undefined` will be passed.
+   */
+  onChange?: (item?: ItemInput) => unknown
+}
+
+/**
+ * A `DropdownMenu` provides an anchor (button by default) that will open a floating menu of selectable items.  The menu can be
+ * opened and navigated using keyboard or mouse.  When an item is selected, the menu will close and the `onChange` callback will be called.
+ * If the default anchor button is used, the anchor contents will be updated with the selection.
+ */
+export function DropdownMenu({
+  renderAnchor = <T extends DropdownButtonProps>(props: T) => <DropdownButton {...props} />,
+  renderItem = Item,
+  placeholder,
+  selectedItem,
+  onChange,
+  ...listProps
+}: DropdownMenuProps): JSX.Element {
+  const anchorRef = useRef<HTMLElement>(null)
+  const [overlayRef, updateOverlayRef] = useRenderForcingRef<HTMLDivElement>()
+
+  const anchorId = `dropdownMenuAnchor-${randomId()}`
+
+  const [open, setOpen] = useState<boolean>(false)
+  const [focusType, setFocusType] = useState<null | 'anchor' | 'list'>(null)
+  const onDismiss = useCallback(() => {
+    setOpen(false)
+    setFocusType(null)
+  }, [])
+
+  const onAnchorKeyDown = useCallback(
+    (event: React.KeyboardEvent<HTMLElement>) => {
+      if (!event.defaultPrevented) {
+        if (!open) {
+          if (event.key === 'ArrowDown' || event.key === 'ArrowUp') {
+            setFocusType('list')
+            setOpen(true)
+            event.preventDefault()
+          } else if (event.key === ' ' || event.key === 'Enter') {
+            setFocusType('anchor')
+            setOpen(true)
+            event.preventDefault()
+          }
+        } else if (focusType === 'anchor') {
+          if (['ArrowDown', 'ArrowUp', 'Tab', 'Enter'].indexOf(event.key) !== -1) {
+            setFocusType('list')
+            event.preventDefault()
+          } else if (event.key === 'Escape') {
+            onDismiss()
+            event.preventDefault()
+          }
+        }
+      }
+    },
+    [open, focusType, onDismiss]
+  )
+  const onAnchorClick = useCallback(
+    (event: React.MouseEvent<HTMLElement>) => {
+      if (!event.defaultPrevented && event.button === 0 && !open) {
+        setOpen(true)
+        setFocusType('anchor')
+      }
+    },
+    [open]
+  )
+
+  const {position} = useAnchoredPosition({anchorElementRef: anchorRef, floatingElementRef: overlayRef})
+
+  useFocusZone({containerRef: overlayRef, disabled: !open || focusType !== 'list' || !position})
+  useFocusTrap({containerRef: overlayRef, disabled: !open || focusType !== 'list' || !position})
+
+  return (
+    <>
+      {renderAnchor({
+        ref: anchorRef,
+        id: anchorId,
+        'aria-labelledby': anchorId,
```

Comment:
Why is the anchor's `aria-labelledby` referencing itself? Is this intentional?

Reply:
Looks like it is accurate to label the button with it's own id based on https://www.w3.org/TR/wai-aria-practices/examples/listbox/listbox-collapsible.html.  If there is an outside label, it should be mixed in first as is done in the story: https://github.com/primer/components/pull/1135/files#diff-14de3214a329b0c98400ce1c0a330f58e8cc55fcb14981be4b1e0951ee556cc2R40

Reply:
TIL!